sql/sys_vars.cc

```diff
@@ -1936,7 +1936,7 @@ static Sys_var_ulong Sys_max_prepared_stmt_count(
        "max_prepared_stmt_count",
        "Maximum number of prepared statements in the server",
        GLOBAL_VAR(max_prepared_stmt_count), CMD_LINE(REQUIRED_ARG),
-       VALID_RANGE(0, 1024*1024), DEFAULT(16382), BLOCK_SIZE(1),
+       VALID_RANGE(0, 1024*1024*1024L), DEFAULT(16382), BLOCK_SIZE(1),
```

Comment:
Any particular reason not to make it `ULONG_MAX`?

Reply:
It is different on 32 and 64bit systems on Unix, but not different onWindows, ULONG_MAX is probably the worst range upper limit we currently have, in terms of portability. it is even hard to put into documentation

Reply:
I'm not big fan of `ulong` either. What about `UINT_MAX32`?

Reply:
sounds good to me.  should not variable itself be int then